frontend/src/metabase/query_builder/components/filters/FilterPopoverHeader.jsx

```diff
@@ -47,6 +47,7 @@ export default function FilterPopoverHeader({
             "ml-auto": !showOperatorSelectorOnOwnRow,
             my1: showOperatorSelectorOnOwnRow,
           })}
+          field={field}
           operator={filter.operatorName()}
           operators={filter.filterOperators()}
```

Comment:
I think it'd be better to pull the logic around text fields out of the UI. Two half-baked ideas for that:
1. Thread the current value down through `filterOperators()`, so that the code generating the list of operators for a given field type can make use of the current value.
2. Update this component to always display the current value in the list even if it isn't one of the `operators`.

Reply:
I'm having a little trouble with option 1 there - `filterOperators` ends up memoized, which seems to not like an `undefined` or empty key for an argument.

Reply:
wait, I figured it out :smile:

Reply:
It looks like `memoize` expected a consistent number of arguments from the caller (ignoring any default arg values). I updated it to use the number of arguments as part of the cache key, which I think fixes the issue. I pushed that change on this branch.